File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.rb
# @param {Integer} num
# @return {Integer}
def number_of_steps(num)
    count = 0
    num_to_reduce = num.to_i
    while num_to_reduce > 0
        if num_to_reduce % 2 != 0 
            num_to_reduce = (num_to_reduce - 1)
            count = count += 1
        else
            num_to_reduce = num_to_reduce / 2
            count += 1
        end
    end
    count - 1
    return count    
end

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.ts
function numberOfSteps(num: number): number {
    let count = 0
    while (num > 0) {
        if (num % 2 !== 0) {
            num--
            count++
        } else {
            num = num / 2
            count++
        }
    }
    return count
};